project/src/main/webapp/emailParse.js

```diff
@@ -30,59 +29,149 @@ class MLModelEmailParse {
+        btnEmail.setAttribute("class", "collapsible");
+        btnEmail.innerHTML = answerText;
+        document.getElementById("emailAnswers").appendChild(btnEmail);
+    }
+
+    extractEmailBodiesToArray(emailDict) {
         const allPass = [];
-        for (let key in exampleDict) {
-            let body = exampleDict[key]["emailBody"];
+        for (let key in emailDict) {
+            let body = emailDict[key]["emailBody"];
             allPass.push(body);
         }
         return allPass;
     }
 
-    /** Create map of scores with values as ML answer and email body. */
-    getScoreToEmail(mlValues, allPass) {
+    getEmailProps(emailObjects, emailBodyValue) {
+        const emailProps = [];
+        for (let key in emailObjects) {
+            let body = emailObjects[key]["emailBody"];
+            if (body == emailBodyValue) {
+                emailProps.push(emailObjects[key]["emailDate"]);
+                emailProps.push(emailObjects[key]["emailSubject"]);
+                emailProps.push(emailObjects[key]["emailSender"]);
+            }
+        }
+        return emailProps;
+    }
+
+    /** Create map of scores with values as ML answer and email properties. */
+    getScoreToEmail(mlValues, allPass, emailObjects) {
         const confidenceWithEmailBody = new Map();
         for (let i = 0; i < mlValues.length; i++) {
             if (mlValues[i].length != 0) {
                 let mlAnswer = mlValues[i][0]["text"];
-                confidenceWithEmailBody.set(mlValues[i][0]["score"], { "answer": mlAnswer, "emailBody": allPass[i] });
+                let arrayEmailProperties = this.getEmailProps(emailObjects, allPass[i]);
+                confidenceWithEmailBody.set(mlValues[i][0]["score"], {
+                    "answer": mlAnswer,
+                    "emailBody": allPass[i],
+                    "emailDate": arrayEmailProperties[0],
+                    "emailSubject": arrayEmailProperties[1],
+                    "emailSender": arrayEmailProperties[2],
+                });
             }
         }
         return confidenceWithEmailBody;
     }
 
-    highestConfidence(answerDict) {
+    arrayOfConfidence(answerDict) {
         let confidenceArray = [];
         for (let key of answerDict.keys()) {
             confidenceArray.push(key);
         }
-        let bestAnswer = confidenceArray.sort((a, b) => b - a);
-        return bestAnswer[0];
+        let orderedArray = confidenceArray.sort((a, b) => b - a);
+        return orderedArray;
+    }
+
+    highlightAnswer(emailBody, answer, bodyTag) {
+        bodyTag.innerHTML = emailBody;
+        let innerHTML = bodyTag.innerHTML;
+        let index = innerHTML.indexOf(answer);
+        if (index >= 0) {
+            innerHTML = "Body: ".bold().fontsize(4) + innerHTML.substring(0, index) + "<span class='highlight'>" + innerHTML.substring(index, index + answer.length) + "</span>" + innerHTML.substring(index + answer.length);
```

Comment:
Break this into a few intermediate steps/variables and use template strings. Something like:

```
const bodyLabel = "Body: ".bold().fontsize(4);
const highlightedText = `<span class='highlight'>${innerHTML.substring(index, index + answer.length)}</span>`;
innerHTML = `${bodyLabel} ${innerHTML.substring(0, index)} ${highlightedText} ${innerHTML.substring(index + answer.length)}`;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals